### PR TITLE
Hamming Distance in c++

### PR DESCRIPTION
**Pull request by aakriti1318**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
### Description

Given x and y, return the number of positions where their values differ in their binary representations as a 32-bit integer.

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2221

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@aakriti1318 Move this program to the math section. Add problem link. Also, we can do this in constant time so you can add this program too along with the existing program

```c++
int solve(int x, int y) {
 int count = 0;
 for (int i = 0; i < 32; i++)
 if ( (x & (1 << i)) != (y & (1 << i)) )
 count++;
 return count;
}
```

File: C-Plus-Plus/bits_manipulation/Hamming_Distance.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Hamming Distance
+/*
+For given x and y, return the number of positions where their values differ in their binary representations as a 32-bit integer.
+
+Input
+x = 10
+y = 6
+Output
+2
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/bits_manipulation/Hamming_Distance.cpp]: Shift sample I/O to the end of the Code

File: C-Plus-Plus/bits_manipulation/Hamming_Distance.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+int solve(int x, int y) {
+    int ans = x^y;
+    return __builtin_popcount(ans); // O(N) time complexity
+}
+
+int main(){
+    int x,y; cin>>x>>y;
+    cout<<solve(x,y)<<endl;
+}
+
```
Comment (HarshCasper) [inline C-Plus-Plus/bits_manipulation/Hamming_Distance.cpp]: Remove the extra line at the EOF

File: C-Plus-Plus/bits_manipulation/Hamming_Distance.cpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using namespace std;
+
+int solve(int x, int y) {
+    int ans = x^y;
+    return __builtin_popcount(ans); // O(N) time complexity
+}
+
+int main(){
+    int x,y; cin>>x>>y;
+    cout<<solve(x,y)<<endl;
```
Comment (HarshCasper) [inline C-Plus-Plus/bits_manipulation/Hamming_Distance.cpp]: Indent the Code and format it properly

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**